### DISCUSSION TITLE
Ko-fi integration for the plugin store

### DISCUSSION THREAD

**Discussion by lukeyouell** [craftcms/console#81] [Plugin Store]
I think that it would be a good idea to allow plugin developers to provide their ko-fi account details so that users have the option to support the development of free plugins in the form of purchasing a coffee.

This could display a button next to 'Install' labelled something like 'Buy Coffee'

Just an idea!

**Comment by khalwat** [comment]
There's also:

https://beerpay.io
https://www.patreon.com

...and a number of others. I think it probably makes sense for the developer to put whatever they want on their github/docs/web pages and not have P&T get involved. There are just too many options people might want.

**Comment by lukeyouell** [comment]
@khalwat That's true! Maybe it would be a good idea to ask developers for their profile links for these services from id.craftcms.com?

The introduction of the plugin store means that users are less likely to visit the Github pages of plugins that don't really require much documentation.

**Comment by carlcs** [comment]
Instead of integrating a third party service, I would rather see a donation feature built right into the Plugin Store.

Quoting my comment from the [Official Plugin Store issue](https://github.com/craftcms/cms/issues/808#issuecomment-276266491):

> It would be nice if the plugin store would allow donations or “pay what you want” licences.
> 
> With some kind of indication in the settings / plugins view that reminds the users that they’ve done good (a small Happy Developer badge or something like this).
> 
> I think this would be a nice way to support developers who
> 
> - want as many people as possible use the plugin (do something good for the Craft platform and community)
> - offer a plugin that make the developer’s life easier (hard sell because the client has no direct benefits from it)
> - offer a small utility/helper plugin (e.g. stop developers abusing Twig)
>
> All these plugins do still require maintenance and support, and I’m quite confident that there’s quite a few developers and client’s who are willing do give something.

Making use of the Store infrastructure would make it easier to donate to developers, as it requires no extra registration to a service. It would also allow to integrate features like the “Happy Developer badge” I mentioned above, which could create an additional incentive to donate.
